beater/server.go

```diff
@@ -133,31 +100,34 @@ func newServer(
 	reporter publish.Reporter,
 	sourcemapStore *sourcemap.Store,
 	batchProcessor model.BatchProcessor,
-) (server, error) {
+) (*server, error) {
 	agentcfgFetchReporter := agentcfg.NewReporter(agentcfg.NewFetcher(cfg), batchProcessor, 30*time.Second)
 	ratelimitStore, err := ratelimit.NewStore(
 		cfg.AgentAuth.Anonymous.RateLimit.IPLimit,
 		cfg.AgentAuth.Anonymous.RateLimit.EventLimit,
 		3, // burst multiplier
 	)
 	if err != nil {
-		return server{}, err
+		return nil, err
 	}
+	// TODO: If tls config changes, should the server restart itself? or
+	// does the serverRunner notice the change and create a new server?
+	// Probably would result in simpler code if the serverRunner handled it.
 	httpServer, err := newHTTPServer(
 		logger, info, cfg, tracer, reporter, batchProcessor, agentcfgFetchReporter, ratelimitStore, sourcemapStore,
 	)
 	if err != nil {
-		return server{}, err
+		return nil, err
 	}
 	grpcServer, err := newGRPCServer(logger, cfg, tracer, batchProcessor, httpServer.TLSConfig, agentcfgFetchReporter, ratelimitStore)
```

Comment:
grpcServer and jaegerServer still need to be checked out to see what can be reloaded vs. restarted